### PR TITLE
Add `vue-template-es2015-compiler` package types

### PR DESCRIPTION
**Pull request by iam-medvedev**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by esafev — APPROVED** [review]

**Comment by jakebailey** [comment]
Does this need to be on DT? It's a 5 year old internal package for Vue. It doesn't feel like anyone would be using it directly...

**Comment by jakebailey** [comment]
Hmmm, I guess some people do need it directly?

**Comment by iam-medvedev** [comment]
This is a low-level package for building vue 2 components. We use this package inside our custom loader for legacy components. I think many people do the same (if we look at the popularity of this package)

File: types/vue-template-es2015-compiler/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Type definitions for vue-template-es2015-compiler 1.9
+// Project: https://github.com/vuejs/vue-template-es2015-compiler#readme
+// Definitions by: Ilya Medvedev <https://github.com/iam-medvedev>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface TransformsOptions {
+    modules?: boolean;
```
Comment (jakebailey) [inline types/vue-template-es2015-compiler/index.d.ts:7]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

Comment (jakebailey) [reply types/vue-template-es2015-compiler/index.d.ts:7]: (and below)

Comment (iam-medvedev) [reply types/vue-template-es2015-compiler/index.d.ts:7]: Make sense, updated.

**Review by jakebailey — APPROVED** [review]

**Review by esafev — APPROVED** [review]

**Comment by iam-medvedev** [comment]
Ready to merge

**Pull request merged**